Makefile

```diff
@@ -45,6 +45,22 @@ sle15_activation: check
 		> dist/suse-migration-sle15-activation.spec
 	tar -czf dist/suse-migration-sle15-activation.tar.gz grub.d
 
+pre_checks: check
+	rm -f dist/*
+	# update rpm changelog using reference file
+	helper/update_changelog.py \
+		--since package/suse-migration-pre-checks.changes.ref \
+		--from package/suse-migration-pre-checks-spec-template --utc > \
+		dist/suse-migration-sle15-activation.changes
+	helper/update_changelog.py \
+		--file package/suse-migration-pre-checks.changes >> \
+		dist/suse-migration-pre-checks.changes
+	# update package version in spec file
+	cat package/suse-migration-pre-checks-spec-template \
+		| sed -e s'@%%VERSION@${version}@' \
+		> dist/suse-migration-pre-checks.spec
+	tar -czf dist/suse-migration-pre-checks.tar.gz suse_migration_services/prechecks
```

Comment:
I'm afraid this is not going to work that way. You are trying to rip-out parts of the python namespace "suse_migration_services" into an extra package. This will fail as soon as the pre-checks code uses anything else from this namespace (for example the logging)

I think the approach with a complete new spec file should be revised. My idea was to use the same source tarball and spec from ```suse-migration-services-spec-template``` and create a sub-package. The sub-package can package the full sources but only the pre-checks entry points as callable tools. This will end up in a package conflict between suse-migration-services and suse-migration-pre-checks because they will partially provide the same files. But this conflict is not an issue and actually also a good thing. By no means the pre-checks will be installed with the migration-services on the same host. 

So I think instead of touching the Makefile just touch the ```suse-migration-services-spec-template``` and create a sub-package definition in there

Reply:
Code has been updated after our conversation.

I wonder if we want to have pre-checks available for `run_migration` as well
it seems to me currently that wouldn't be possible, IMHO it should be possible.

Thoughts ?

Reply:
This is a bit of a problem. If customer decides to go with the real reboot mode, the package suse-migration-sle15-activation gets installed and the pre-checks runs as part of %post. So all good in this case.

If the customer decides to go with run_migration, no activation package will be installed because run-migration is provided with the package that contains the migration live image (SLE15-Migration). In this case no pre-checks are available

If the customer installs suse-migration-sle15-activation for the reason to get the pre-checks, there will also be the grub menu entry which is unwanted but does not hurt.

For the moment we could add some information in the documentation that says to run the pre-checks, install the
activation package. No matter how the migration is started, the activation and also migration packages will be uninstalled by the migration code itself (except this stage is not reached by the DMS)

Another option would be to split the suse-migration-sle15-activation package into two packages:

* suse-migration-sle15-activation 
* suse-migration-sle15-pre-checks

The activation package would require the pre-checks, such that %post works. Customers using run_migration can install only the pre-checks package and manually call them. This needs to be documented. And a new pre-checks package must be published through the SUSE process

No matter what we do here, this should be done in a next step, not in this request.

Thoughts ?

Reply:
I have added #209 to tackle this one